freshmaker/errata.py

```diff
@@ -124,6 +124,25 @@ def from_advisory_id(cls, errata, errata_id):
             product_data["product"]["short_name"], cve_list,
             has_hightouch_bug)
 
+    def is_flatpak_module_advisory_ready(self):
+        """ Returns True only if a Flatpaks can be rebuilt from module advisory.
+
+        Flatpaks can be rebuilt only if all of the following are true:
+        - Advisory must contain modules.
+        - Advisory is in QE state.
+        - All attached builds are only shipped to hidden pulp repositories.
+        - All attached builds are signed.
+        """
+        errata = Errata()
+        return (
+            self.state == "QE" and
+            "module" in self.content_types and
+            all(
+                "-hidden-" in repo_id for repo_id in errata.get_pulp_repository_ids()
+            ) and
+            errata.builds_signed(self.errata_id)
```

Comment:
Interesting, I was not aware there is already a `builds_signed` api, maybe we can use update it to the new errata API (RHELWF-4641) which will be released soon, it will be more efficient because it doesn't need to call errata API for each build. However this can be optional change in this PR, the old `builds_signed` check is actually unnecessary and the related event and handler can be refactored (mentioned in CLOUDWF-6150).